packages/mds-geography-service/service/provider.ts

```diff
@@ -117,6 +118,18 @@ export const GeographyServiceProvider: ServiceProvider<GeographyService, Geograp
     }
   },
 
+  deleteGeographyAndMetadata: async (context, geography_id) => {
+    try {
+      await GeographyRepository.deleteGeography(geography_id)
+      await GeographyRepository.deleteGeographyMetadata(geography_id)
```

Comment:
Ah, I see, we don't call `deleteGeographyMetadata` until after calling `deleteGeography` successfully. Could we enforce this somehow?

Reply:
It's better to let the Service define the rules for interaction between data models. 
Like, when lazy, i had a "assert geography exist" check in the "update metadata" repo function. But that limits the rules that you can write on those actions, when all possible features are allowable inside the boundaries of the data service.